bin/lint

```diff
@@ -2,22 +2,32 @@
 
 set -eu
 
+lintversion=1.14.0
+
 cd "$(pwd -P)"
 
 bindir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 rootdir="$( cd $bindir/.. && pwd )"
 
 cd "$rootdir"
 
-# install golint from vendor
-go install ./vendor/golang.org/x/lint/golint
+os=linux
+exe=
+if [ "$(uname -s)" = "Darwin" ]; then
+  os=darwin
+elif [ "$(uname -o)" = "Msys" ]; then
+  os=windows
+  exe=.exe
+fi
 
-# use `go list` to exclude packages in vendor
-out=$(go list ./... | xargs golint) || true
+lintbin="${rootdir}/.golangci-lint-${lintversion}${exe}"
 
-if [ -n "$out" ]; then
-  echo "$out"
-  exit 1
+if [ ! -f "$lintbin" ]; then
+  curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/v$lintversion/install.sh | sh -s -- -b . v$lintversion
+  mv ./golangci-lint${exe} $lintbin
 fi
 
-echo "all clean!"
+$lintbin run \
+  --enable gofmt,golint \
+  --disable deadcode,errcheck,gosimple,staticcheck,structcheck,unused,varcheck \
+  "$@"
```

Comment:
Am not sure exactly what's going on, but `bin/lint` no longer detects missing comments on publicly-exported structs and functions, which it previously caught.

I've modified my local branch as follows:

```diff
diff --git a/controller/proxy-injector/webhook.go b/controller/proxy-injector/webhook.go
index 1a25ff0d..5a55fae2 100644
--- a/controller/proxy-injector/webhook.go
+++ b/controller/proxy-injector/webhook.go
@@ -25,9 +25,6 @@ const (
 	envVarKeyProxyID                    = "LINKERD2_PROXY_ID"
 )
 
-// Webhook is a Kubernetes mutating admission webhook that mutates pods admission
-// requests by injecting sidecar container spec into the pod spec during pod
-// creation.
 type Webhook struct {
 	client              kubernetes.Interface
 	deserializer        runtime.Decoder
```

When I run `bin/lint`, I get:

```bash
$ bin/lint 
$ echo $?
0
```

Whereas if I run the old command, I get:

```
$ go list ./... | xargs golint
/Users/kl/workspace/go/src/github.com/linkerd/linkerd2/controller/proxy-injector/webhook.go:28:6: exported type Webhook should have comment or be unexported
```

I guess they're using different version of golint?

Reply:
Oh wow nice catch. Looking.